addons/website/static/src/snippets/s_facebook_page/000.js

```diff
@@ -24,10 +22,39 @@ const FacebookPageWidget = publicWidget.Widget.extend({
             params.href = `https://www.facebook.com/${params.id}`;
         }
         delete params.id;
+
+        this._renderIframe(params);
+
+        this.resizeObserver = new ResizeObserver(debounce(this._renderIframe.bind(this, params), 100));
+        this.resizeObserver.observe(document.querySelector("#wrapwrap"));
+
+        return def;
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        this._super.apply(this, arguments);
+
+        this.options.wysiwyg && this.options.wysiwyg.odooEditor.observerUnactive();
+        if (this.iframeEl) {
+            this.iframeEl.remove();
+        }
+        this.options.wysiwyg && this.options.wysiwyg.odooEditor.observerActive();
+        this.resizeObserver.disconnect();
+    },
+    /**
+     * _renderIframe: prepare iframe element & replace it with existing iframe
+     *
+     * @private
+    */
+    _renderIframe: function (params) {
+        this.options.wysiwyg && this.options.wysiwyg.odooEditor.observerUnactive();
+
         params.width = utils.confine(Math.floor(this.$el.width()), 180, 500);
```

Comment:
the max width is 500, so if the previous width was 500 and the new one is also 500 after this line. you don't need to refresh the iframe.